[FEAT] Implementation of integration settings with Activepieces AI Agent for automation

RolandM99 opened the pull request:
# PR

- [ ] Have you followed the [contributing guidelines](https://github.com/ever-co/ever-gauzy/blob/master/.github/CONTRIBUTING.md)?
- [ ] Have you explained what your changes do, and why they add value?

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

---

evereq commented:
@RolandM99 " RolandM99 deleted a comment from coderabbitai bot 3 days ago" why? I.e. why you deleting comments? We should "Resolve" conversations, not delete anything. If you like what AI suggest, please do it and "Resolve". If you don't like, it's REQUIRED to write comment and explain why AI suggestion does not make sense to apply.

RolandM99 commented:
@evereq , You're right ...
I deleted them because I changed the flow. As the PR was still on draft, I chose to delete those comments—my bad!😞 
But this will not happen anymore.
PS: the deleted comments don't affect the updated code.

RolandM99 commented on packages/plugins/integration-activepieces/src/lib/activepieces-authorization.controller.ts:
```diff
@@ -0,0 +1,134 @@
+	 *
+	 * @returns {string} Random state string
+	 */
+	private generateState(): string {
+		return Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
+	}
```
@greptile-apps, no need to encrypt the state here, the CSRF is enough

RolandM99 commented on packages/plugins/integration-activepieces/src/lib/activepieces.service.ts:
```diff
@@ -0,0 +1,315 @@
+					createdAt: new Date().toISOString()
+				},
+				type: ActivepiecesConnectionType.OAUTH2,
+				value: {
+					type: ActivepiecesConnectionType.OAUTH2,
+					client_id: clientId,
+					client_secret: clientSecret,
+					data: {
+						tenantId,
+						organizationId: organizationId || 'default'
+					}
+				}
```
@coderabbitai, according to this documentation of activepieces in the following link: https://www.activepieces.com/docs/endpoints/connections/upsert#o-auth2 , it show how we need to pass OAuth client credentials as value.
